src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -146,14 +149,18 @@ internal class HttpManager : IHttpManager
             return response;
         }
 
-        private HttpClient GetHttpClient(X509Certificate2 x509Certificate2)
-        {
-            if (_httpClientFactory is IMsalMtlsHttpClientFactory msalMtlsHttpClientFactory)
+        private HttpClient GetHttpClient(X509Certificate2 x509Certificate2, HttpClient customHttpClient) {
+            if (x509Certificate2 != null && _httpClientFactory is IMsalMtlsHttpClientFactory msalMtlsHttpClientFactory)
             {
                 // If the factory is an IMsalMtlsHttpClientFactory, use it to get an HttpClient with the certificate
                 return msalMtlsHttpClientFactory.GetHttpClient(x509Certificate2);
             }
 
```

Comment:
if both certificate and customHttpClient are configured, I would add a `Debug.Assert("Not supported. Ignoring the customHttpClient`) and log.error(). Or throw an exception like NotImplementedException